From fb864ced397b4e61b042f65e485053c587d1db04 Mon Sep 17 00:00:00 2001
From: Naveen <nagu@cypress.com>
Date: Tue, 6 Feb 2018 11:52:58 +0530
Subject: [PATCH 016/154] brcmfmac: do not print ulp_sdioctrl get error

Don't print ulp_sdioctrl get error as errors are expected for non ulp cases.


Signed-off-by: Naveen Gupta <nagu@cypress.com>

---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c | 12 ++----------
 1 file changed, 2 insertions(+), 10 deletions(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c
index b32912c..baaacab 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c
@@ -2567,23 +2567,15 @@ static int brcmf_sdio_intr_rstatus(struct brcmf_sdio *bus)
 	struct brcmf_bus *bus_if = dev_get_drvdata(dev);
 	struct brcmf_sdio_dev *sdiodev = bus_if->bus_priv.sdio;
 	struct brcmf_if *ifp = bus_if->drvr->iflist[0];
-	int err = 0;
-	char iovbuf[BRCMF_DCMD_SMLEN];
 
 	brcmf_dbg(TRACE, "Enter\n");
-	memset(iovbuf, 0, sizeof(iovbuf));
 
 	/* Query ulp_sdioctrl iovar to get the ULP related SHM offsets */
-	err = brcmf_fil_iovar_data_get(ifp, "ulp_sdioctrl", iovbuf,
-				       sizeof(iovbuf));
-	if (err)
-		brcmf_err("fail to get ulp_sdioctrl err:%d\n", err);
+	brcmf_fil_iovar_data_get(ifp, "ulp_sdioctrl", &sdiodev->shm_ulp,
+				      sizeof(sdiodev->shm_ulp));
 
 	sdiodev->ulp = false;
 
-	/* Copy the data shared by dongle to FMAC structure */
-	memcpy(&sdiodev->shm_ulp, iovbuf, sizeof(struct ulp_shm_info));
-
 	brcmf_dbg(TRACE, "m_ulp_ctrl_sdio[%x] m_ulp_wakeevt_ind [%x]\n",
 		  M_DS1_CTRL_SDIO(sdiodev->shm_ulp),
 		  M_WAKEEVENT_IND(sdiodev->shm_ulp));
-- 
1.9.1

